src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject.cs

```diff
@@ -59,6 +59,15 @@ protected sealed override bool TryGetOutputPathFromHierarchy(out string binOutpu
         protected sealed override bool LastDesignTimeBuildSucceeded => _lastDesignTimeBuildSucceeded;
 
         // We might we invoked from a background thread, so schedule the disconnect on foreground task scheduler.
-        public sealed override void Disconnect() => InvokeBelowInputPriority(() => base.Disconnect());
+        public sealed override void Disconnect()
+        {
+            InvokeBelowInputPriority(() =>
+            {
+                // clear code model cache and shutdown instances, if any exists.
+                _projectCodeModel?.OnProjectClosed();
```

Comment:
Should this just be moved to base.Disconnect? It looks like we do it in all the other derived types too.